pep-0692.rst

```diff
@@ -0,0 +1,438 @@
+PEP: 692
+Title: Using TypedDict for more precise \*\*kwargs typing
+Author: Franek Magiera <framagie@gmail.com>
+Sponsor: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 29-May-2022
+Python-Version: 3.12
+Post-History: 29-May-2022
+
+
+Abstract
+========
+
+Current specification enables for type hinting ``**kwargs`` as long as all of
+the keyword arguments specified by them are of the same type. However, that
+behaviour can be very limiting. Therefore, in this PEP we propose a new way to
+enable more precise ``**kwargs`` typing. The new approach revolves around using
+``TypedDict`` to type ``**kwargs`` that comprise keyword arguments of different
+types.
+
```

Comment:
Might be worth highlighting near the beginning that:

* a Python grammar change is being proposed

Since that is useful context for a reader to determine whether they need to respond to changes in the PEP.

Reply:
It would be nice if a note could be added that

> Usage of Unpack for *args and TypeVarTuple is proposed in PEP 589.

or an update note added to `PEP 589` to the effect of:

> **UPDATE:** Usage of Unpack also for **kwargs and TypedDict is proposed in PEP 692.

to link the two and provide greater traceability. It's a little unclear to me whether or not `Unpack` will work for all 4.

Reply:
`Unpack` isn't mentioned until the Backwards Compatibility section where it is linked to PEP 646. Added a note that the new use cases of `Unpack` should not interfere with the behaviour described in 646.